OMEdit/OMEditGUI/Editors/BaseEditor.cpp

```diff
@@ -731,6 +747,26 @@ void PlainTextEdit::insertCompleterTypes(QStringList types)
     QStandardItem *pStandardItem = new QStandardItem(types[k]);
     pStandardItem->setIcon(QIcon(":/Resources/icons/completerType.svg"));
     pStandardItem->setToolTip("types");
+    CompleterItem completerItem;
+    completerItem.key=types[k];
+    completerItem.value=types[k];
+    pStandardItem->setData(QVariant::fromValue(completerItem),Qt::UserRole);
+    mpStandardItemModel->appendRow(pStandardItem);
+  }
+}
+
+void PlainTextEdit::insertCompleterCodeSnippets(QList<CodeSnippetItem>  items)
+{
+  for (int var = 0; var < items.length(); ++var)
+  {
+    QStandardItem *pStandardItem = new QStandardItem(items[var].mKeyword);
+    pStandardItem->setIcon(QIcon(":/Resources/icons/completerCodeSnippets.svg"));
+    pStandardItem->setToolTip("codeSnippets");
+    CompleterItem completerItem;
+    completerItem.key=items[var].mKeyword;
+    completerItem.value=items[var].mValue;
+    completerItem.select=items[var].mSelection;
+    pStandardItem->setData(QVariant::fromValue(completerItem),Qt::UserRole);
```

Comment:
Also instead of removing the class, I think we can keep the class CompleterItem and remove the struct CompleterItem, as class can also be used with QVariant